### ISSUE TITLE
Support choice parameters?

### ISSUE THREAD

**Issue by benwing** [scallop/scallop#145]
I looked for choice parameters in Scallop and didn't see them. The idea is to add a 'choices' argument to opt[] that takes a collection of values of the right type, and causes an error to be thrown if the value specified isn't one of the possible values. This is how choices are supported in Python's argparse, for example (https://docs.python.org/3/library/argparse.html#choices).

**Comment by Rogach** [comment]
I think you can achieve this using `validate` on the option:

```scala
opt[String](validate = (value) => Set("alpha", "beta", "gamma").contains(value))
```
Or, using the fact that `Set` extends `A => Boolean`:
```scala
opt[String](validate = Set("alpha", "beta", "gamma"))
```

**Comment by ayakovlenko** [comment]
One advantage of choices over validate is that it makes it possible to automatically generate meaningful help and error messages.

**Comment by Rogach** [comment]
@ayakovlenko - That's true. Can you give an example of good help message in this case?

**Comment by ayakovlenko** [comment]
I think something as simple as `Choices: ${choices.mkString("(", ", ", ")")}`, or any other format will do. It can, for example, be appended to the description.

Yes, I'm just copycatting `scalac -help` here:

```
-target:<target> Target platform for object files. All JVM 1.5 - 1.7 targets are deprecated. Choices: (jvm-1.5,jvm-1.6,jvm-1.7,jvm-1.8), default: jvm-1.8.
```

**Comment by Rogach** [comment]
Yes, the format looks good, especially if it is already used in big tools. I'll implement it, hopefully on this week.

**Comment by Rogach** [comment]
Released `3.1.0` with this change, should sync to Maven Central within an hour.

The syntax looks as follows:
```scala
val letter = choice(Seq("alpha", "beta", "gamma"))
```
